### PR TITLE
feat: show group info in users list

### PR DESCRIPTION
**Pull request by magnew**
Closes: #8535 

### Description:

Shows basic group info in the users list. The number of groups is in the list and the groups are shown on hover.

Screenshot 2024-01-10 at 19 08 25

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/common/src/types/organizationMemberProfile.ts [hunk 1]
```diff
@@ -43,6 +43,10 @@ export type OrganizationMemberProfileWithGroups = OrganizationMemberProfile & {
     groups: Pick<Group, 'name' | 'uuid'>[];
 };
 
+export const isOrganizationMemberProfileWithGroups = (
+    obj: OrganizationMemberProfile | OrganizationMemberProfileWithGroups,
+): obj is OrganizationMemberProfileWithGroups => 'groups' in obj;
```
Comment (IrakliJani) [inline packages/common/src/types/organizationMemberProfile.ts:48]: 👍

File: packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/UsersView.tsx [hunk 2]
```diff
@@ -275,10 +324,13 @@ const UsersView: FC = () => {
...
     const { classes } = useTableStyles();
 
+    // TODO: this is a feature flag while we are building groups.
+    const groupManagementEnabled = useFeatureFlagEnabled('group-management');
+
     const [search, setSearch] = useState('');
 
     const { data: organizationUsers, isLoading: isLoadingUsers } =
-        useOrganizationUsers(search);
+        useOrganizationUsers({ searchInput: search, includeGroups: 1000 });
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/UsersView.tsx:333]: is 1000 intentional?

**Review by IrakliJani — APPROVED** [review]
looks good, few comments - not a blocker

**Comment by IrakliJani** [comment]
tests failing because I think seed did not run? resetting db

File: packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/UsersView.tsx (hunk 2 above)
Comment (magnew) [reply packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/UsersView.tsx:333]: Yea, but I'll put a TODO. And maybe make it 10K. ATM these endpoints require a number, but we aren't doing pagination

**Pull request merged**